src/timer.rs

```diff
@@ -77,7 +77,7 @@ macro_rules! hal {
                     T: Into<Hertz>,
                 {
                     // enable and reset peripheral to a clean slate state
-                    apb1.enr().write(|w| w.$timXen().set_bit());
+                    apb1.enr().write(|w| w.$timXen().set_bit()); // AJM - why `write` and not `modify` here?
```

Comment:
This is unchanged from stm32f30x (not implemented on my chip yet) but I'm guess that is a write-only register and hence the R/M/W is unnecessary.